cscs-checks/libraries/io/netcdf_compile_run.py

```diff
@@ -43,9 +43,16 @@ def __init__(self, lang, linkage):
 
     @rfm.run_before('compile')
     def setflags(self):
-        environ = self.current_environ
+        # FIXME: static compilation yields a link error in case of
+        # PrgEnv-cray(Cray Bug #255707)
+        if (self.linkage == 'static' and
+            self.current_system.name == 'dom' and
+            self.current_environ.name == 'PrgEnv-cray'):
+            self.variables = {'LINKER_X86_64': '/usr/bin/ld',
+                              'LINKER_AARCH64': '=/usr/bin/ld'}
```

Comment:
Put everything in a separate decorated function. ReFrame will execute all the hooks. Use the same name as the one proposed above.